pebblo/app/service/local_ui_service.py

```diff
@@ -367,3 +380,57 @@ def get_vector_dbs(chains):
         for data in chains:
             vector_dbs.extend([db["name"] for db in data["vectorDbs"]])
         return list(set(vector_dbs))
+
+    @staticmethod
+    def get_all_documents(retrieval_data):
+        """This function returns documents per app"""
+        documents = {}
+        all_sorted_documents = {}
+
+        for data in retrieval_data:
+            data_context = data.get("context")
+            for context in data_context:
+                document_name = context["retrieved_from"]
+                if document_name in documents.keys():
+                    documents[document_name].extend([data])
+                else:
+                    documents[document_name] = [data]
+
+        all_documents = sorted(
+            documents.items(), key=lambda kv: (len(kv[1]), kv[0]), reverse=True
+        )
+
+        for user_name, data in all_documents:
```

Comment:
It would be better if you could add single line comment at 2 places in this function about what each section is doing.
For example:
```
# Fetch documents from all retrievals.
# Sort documents based on number of times they used in the retrieval.
# Prepare sorted dictionary based on number of times they used in the retrieval.
```